### DISCUSSION TITLE
about the run of this program

### DISCUSSION THREAD

**Discussion by Duccu3** [manucabral/YoutubeMusicRPC#16] [Q&A]
1) Do i have to open the cmd all the time to active it ?
2) It doesn't recognize my desktop app and always my opera browser ? how can i make it possible in my youtube music desktop app ?

**Comment by Tovaryon** [comment]
[IM NOT THE MAIN CONTRIBUTANT SO I CAN SAY SOMETHING WRONG!]

Hello Duccu3.
As i know you cant use the program as a background task yet since its not programmed for that and i think that the Desktop App does not work with the program but you theoretically could use the "desktop" YTMusic App from Chrome.

Shuckes | LittlePengluin

**Reply by Duccu3** [reply]
i use opera gx , do opera gx have the "desktop"  app like the chrome counterpart ?

**Reply by Tovaryon** [reply]
No. Only chrome has that

**Comment by Duccu3** [comment]
and i also have a question that sometime the program doesnt show the current song . How can i fix that ?

**Reply by Tovaryon** [reply]
Try pressing Enter

**Comment by Duccu3** [comment]
And again . It happen when i switched to M Edge and the program still only detected Opera , how can i fix that ?

**Reply by Tovaryon** [reply]
You have to make the browser your main browser. Not just open the browser.

**Comment by manucabral** [comment]
> 

Hi,
1. The latest version of the application initiates a cmd that displays some info for you, and also you can close it from there. In the next update, it won't be like this anymore.

2. The application recognizes your default browser, which is why it always opens Opera in your case.


**Some browsers allow you to install the YouTube Music application. I'll provide you with an example gif.**
![test](https://github.com/manucabral/YoutubeMusicRPC/assets/83357673/be909f1d-ceee-4052-817c-59fa6cde8932)

You need to find out with which browser you've installed the application; by default, it's always Microsoft Edge. Once you find it, you just have to set that browser as the default browser

**I also provide you with an example of how to use the application once you've set the default browser. (using Youtube Music Desktop  App installed from Google Chrome)**
![test2](https://github.com/manucabral/YoutubeMusicRPC/assets/83357673/77851836-f6c4-4501-9eea-3003eb7d4eb2)


I hope I've been helpful. If you need anything else, I'm here and ready to assist u!

**Reply by Duccu3** [reply]
![image](https://github.com/manucabral/YoutubeMusicRPC/assets/142346260/7d139013-eeec-4b01-9027-8079d2445feb)
when i changed my default to Edge and install the Desktop App , i do just like what u did above but it still show error .
How can i fix it ?

**Reply by manucabral** [reply]
If you are using Edge, first close all Edge processes in the task manager, and then try it again.
Once you've opened YoutubeMusicRPC, the Edge browser will open, and do not close it. Simply open the YouTube Music Desktop App.

![image](https://github.com/manucabral/YoutubeMusicRPC/assets/83357673/9161725c-dee7-4ad3-994f-caf630cc6529)

**Comment by Cszyn** [comment]
hey so what is client id? and where can i find it?

**Reply by Cszyn** [reply]
![Youtube_Music_Rich_Presence_w3wLf8dTUo](https://github.com/manucabral/YoutubeMusicRPC/assets/147027531/7efda391-9de5-4c91-b92e-eb359d903c55)

**Reply by manucabral** [reply]
The clientID is used for Discord to identify the application. Any user can create one and name it as they wish, but in this case, you simply need to specify that you will not be using a custom clientID.

1. Delete the "settings.json" file.
2. Run the application.
3. Type "no" when prompted for a custom client ID.

![image](https://github.com/manucabral/YoutubeMusicRPC/assets/83357673/c2d33d48-afd4-428b-9d05-1c4b0fe7ca5c)
